tests/integration/identity.go

```diff
@@ -66,30 +92,33 @@ func newIdentityHolder(ident acpIdentity.Identity) *identityHolder {
 
 // getIdentity returns the identity for the given reference.
 // If the identity does not exist, it will be generated.
-func getIdentity(s *state, ref immutable.Option[identityRef]) acpIdentity.Identity {
-	if !ref.HasValue() {
+func getIdentity(s *state, identity immutable.Option[identity]) acpIdentity.Identity {
+	if !identity.HasValue() {
 		return acpIdentity.Identity{}
 	}
-	return getIdentityHolder(s, ref.Value()).Identity
+	if identity.Value().selector == "*" {
+		require.Fail(s.t, "Used the \"*\" selector for identity incorrectly.", s.testCase.Description)
+	}
```

Comment:
todo: Is this safeguarding against using `AllClientIdentities` for a node identity? Either way, please document why you're doing this check because at first glance it looks like a mistake. My first thought was: why am I able to set the selector as `*` but then it will give me an error if the selector is `*`?

Reply:
That is because the "*" selector should never be passed to `getIdentity`. Get identity is used to fetch the identity from the stored identities, while "*" is used to signal to acp that we want to target not a specific `did` but `any` identity.

If you look at `getIdentityDID` function it handles the "*" selector before `getIdentity` is called.

Nonetheless can document this. LMK if have any other suggestions to improve this.

Reply:
Thank you for documenting! It very much removes the confusion 👍